### PR TITLE
Remove node.kubernetes.io/not-ready taint if status active after creation.

### LINKED ISSUE #3709: Node has `node.kubernetes.io/not-ready` taint in integration tests even if the Node is ready
**What happened**:
In the integration tests nodes have `node.kubernetes.io/not-ready` taint even if in their status `Ready` condition is set to `true` 

**What you expected to happen**:
Nodes didn't have the taint

**How to reproduce it (as minimally and precisely as possible)**:
Run newly added integration test from this [PR](https://github.com/kubernetes-sigs/kueue/pull/3708/files#diff-f24b50c55e86bbae151e985e718b4280d2585110e1b692c579a4a0c6c191036d) with toleration deleted 

**Anything else we need to know?**:
I believe we should add util function that removes the taint based on `Ready` condition

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Remove node.kubernetes.io/not-ready taint if status active after creation.

#### Which issue(s) this PR fixes:

Fixes #3709

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/integration/tas/tas_test.go [hunk 1]
```diff
@@ -261,10 +261,7 @@ var _ = ginkgo.Describe("Topology Aware Scheduling", ginkgo.Ordered, func() {
 						Ready().
```
Comment (mimowo) [inline test/integration/tas/tas_test.go:261]: Please drop in this file the not-ready tolerations like these: https://github.com/kubernetes-sigs/kueue/blob/8cc315716e31d8439665452f8e5627136e352363/test/integration/tas/tas_test.go#L635-L638

File: test/util/util.go [hunk 2]
```diff
@@ -795,3 +796,30 @@ func ExpectLocalQueuesToBeActive(ctx context.Context, c client.Client, lqs ...*k
...
+		// Update status based on the object
+		createdNode := &corev1.Node{}
+		gomega.EventuallyWithOffset(1, func(g gomega.Gomega) {
+			g.Expect(c.Get(ctx, client.ObjectKeyFromObject(&node), createdNode)).Should(gomega.Succeed())
+			createdNode.Status = node.Status
+			g.Expect(c.Status().Update(ctx, createdNode)).Should(gomega.Succeed())
+		}, Timeout, Interval).Should(gomega.Succeed())
+		// Removes the taint if the node is Ready
+		gomega.EventuallyWithOffset(1, func(g gomega.Gomega) {
+			if !utiltas.IsNodeStatusConditionTrue(node.Status.Conditions, corev1.NodeReady) {
```
Comment (mimowo) [inline test/util/util.go:813]: I think this should be the other way round - we filter if Ready

Comment (mbobrovskyi) [reply test/util/util.go]: Yes, you right. Thanks!

File: test/integration/tas/tas_test.go (hunk 1 above)
Comment (mbobrovskyi) [reply test/integration/tas/tas_test.go:261]: Done

**Pull request merged**